woocommerce/woocommerce
# Fix unit tests for order and customer functions to provide HPOS compat.

### All Submissions:






### Changes proposed in this Pull Request:



Closes #36684.
Targets #36843.

This PR adds compatibility for `wc_reset_order_customer_id_on_deleted_user` function and fixes the test in [class-wc-tests-order-functions.php](https://github.com/woocommerce/woocommerce/pull/36905/files#diff-f1542b5bc92b6b70872e6af265bbee9d1be017bd1b6ee197a5212756b4239e9f).

For wc_reset_order_customer_id_on_deleted_user, HPOS is supported by adding direct query for HPOS table as well, just as it was being done for the post table. If the sync is on, then the post is also updated right away, since this is a direct query and would not be triggering sync functions otherwise.

The class-wc-tests-order-functions.php](https://github.com/woocommerce/woocommerce/pull/36905/files#diff-f1542b5bc92b6b70872e6af265bbee9d1be017bd1b6ee197a5212756b4239e9f) is also fixed to load the correct data store.






### How to test the changes in this Pull Request:




Unit test passing is enough, however, if you'd like you can test the customer function via following commands:

1. Create an order via admin or via checkout, but associated with a user. For example, if you are using checkout to create the order, make sure that you are logged in as a customer user. When creating via backend, link the order to an existing customer user. Note the ID of the order.
2. Delete the customer user that is used while creating the order by going to Users > All users, then searching for the user, and clicking on delete from the user row.
3. Go back to the order in admin view, it will now be assigned to a guest.



### Other information:




### FOR PR REVIEWER ONLY:
